### PR TITLE
Pract03: Add 04-05 | Presiyana Borisova

### PR DESCRIPTION
**Pull request by PresiyanaBB**
Tasks 04 and 05 solution
Presiyana - 3MI0600240

### CODE REVIEW

File: SI/Sem.03/Pract.03/Task04.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    cout << "Please enter 2 whole positive numbers: ";
+
+    cin >> m >> n;
+
+    for (int i = 1; i <= n*m; i++)
+    {
+        cout << i << " ";
+
+        if (i%n == 0)
```
Comment (Justsvetoslavov) [inline SI/Sem.03/Pract.03/Task04.cpp:17]: if (i % n == 0)

File: SI/Sem.03/Pract.03/Task05.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+using namespace std;
+
+int main()
+{
+    int targetNumber;
+    int counter = 0;
+    int guessNumber;
+
+    cout << "Please enter whole number from 0 to 100: ";
+    cin >> targetNumber;
```
Comment (Justsvetoslavov) [inline SI/Sem.03/Pract.03/Task05.cpp:11]: targetNumber is between 0 and 100 , add a check.

File: SI/Sem.03/Pract.03/Task05.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#include <iostream>;
+using namespace std;
+
+int main()
+{
+    int targetNumber;
+    int counter = 0;
```
Comment (Justsvetoslavov) [inline SI/Sem.03/Pract.03/Task05.cpp:7]: Suggestion: move the int counter = 0; just before the loop.

File: SI/Sem.03/Pract.03/Task05.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+        cin >> guessNumber;
+        counter++;
+
+        if (guessNumber == targetNumber)
+        {
+            cout << "You won with " << counter << " attempts";
+            break;
+        }
+    } while (true);
```
Comment (Justsvetoslavov) [inline SI/Sem.03/Pract.03/Task05.cpp:25]: We are looping until the numbers are equal.
Can we avoid/change the while (true) ?

File: SI/Sem.03/Pract.03/Task05.cpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    cout << "Please enter whole number from 0 to 100: ";
+    cin >> targetNumber;
+
+    do
+    {
+        cout << "Enter your guess: ";
+
+        cin >> guessNumber;
+        counter++;
+
```
Comment (Justsvetoslavov) [inline SI/Sem.03/Pract.03/Task05.cpp:19]: if (guessNumber < targetNumber)
				cout << "The number you are looking for is greater than your guess!\n";
 else if (guessNumber > targetNumber)
				cout << "The number you are looking for is less than your guess!\n";

File: SI/Sem.03/Pract.03/Task05.cpp [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        if (guessNumber == targetNumber)
+        {
+            cout << "You won with " << counter << " attempts";
```
Comment (Justsvetoslavov) [inline SI/Sem.03/Pract.03/Task05.cpp:22]: Should be outside of the loop.
Our condition is in the while (<condition>)

**Review by Justsvetoslavov — CHANGES_REQUESTED** [review]

**Comment by GeorgiTerziev02** [comment]
The correct solutions have already been uploaded
https://github.com/Justsvetoslavov/Introduction_to_programming/tree/master/SI/Sem.03/Pract.03

**Pull request closed without merging**